build-and-test.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -e 	# Exit immediately upon failure
+set -o pipefail  # Carry failures over pipes
+
+repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+docker_repo="microsoft/dotnet-nightly"
+
+function build_dockerfiles {
+    for dockerfile_dir in $( egrep -v 'nanoserver|windowsservercore' <<< "${1}" | sort ); do
+        tag="${docker_repo}:$( sed -e 's/.\///' -e 's/debian\///' -e 's/debian/sdk/' -e 's/\//-/g' <<< "${dockerfile_dir}" )"
+        echo "----- Building ${tag} -----"
+        docker build -t "${tag}" "${dockerfile_dir}"
+    done
+}
+
+pushd "${repo_root}" > /dev/null
+
+build_dockerfiles "$( find . -path './.*' -prune -o -name 'Dockerfile' -a -path '*/*-deps/*' -print0 | xargs -0 -n1 dirname )"
```

Comment:
The `xargs -0 -n1 dirname` logic is duplicated here rather than put in the `build_dockerfiles` function because when I put it in `build_dockerfiles`, it seemed that the null characters separating files from `find [...] -print0` were not preserved, causing the `xargs` call to not work correctly.  I didn't find a way to preserve them.

An alternative approach would be to use `-print` instead of `-print0` and then omit `-0` from the `xargs` call, but my understanding is that in that case, spaces and other special characters in file paths may not be handled correctly, so I decided it was preferable to duplicate the calls to `xargs`, especially because it only occurs twice and is a small amount of logic.  I'm open to suggestions on this, though.